[1단계 - 출석 구현] 시소(이경민) 미션 제출합니다.

lifeishiphop opened the pull request:
<!-- 

## 코드 리뷰 팁

- 코드와 관련된 질문이 있다면, PR 본문에 적기 보다는 해당 코드를 선택하고 코멘트를 남겨주세요.
  - [참고: Adding comments to a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-comments-to-a-pull-request)

-->

## 체크 리스트

- [x] 미션의 필수 요구사항을 모두 구현했나요?
- [x] Gradle `test`를 실행했을 때, 모든 테스트가 정상적으로 통과했나요?
- [x] 애플리케이션이 정상적으로 실행되나요?
- [x] [프롤로그](https://prolog.techcourse.co.kr)에 셀프 체크를 작성했나요?
  - <!-- 작성한 셀프 체크의 링크를 남겨주세요. -->


## 객체지향 생활체조 요구사항을 얼마나 잘 충족했다고 생각하시나요?

### 1~5점 중에서 선택해주세요.

- [ ] 1 (전혀 충족하지 못함)
- [ ] 2
- [x] 3 (보통)
- [ ] 4
- [ ] 5 (완벽하게 충족)

### 선택한 점수의 이유를 적어주세요.

기능 구현에 집중하다 보니 가독성이 떨어지거나, 객체의 책임을 명확히 고려하지 않아 Crew에게 가중된 책임이 부여되는 등 제 주관으로 바라보았을 때 부족한 부분이 있다고 생각해요.
  
## 어떤 부분에 집중하여 리뷰해야 할까요?

<!-- 리뷰어가 효과적으로 피드백할 수 있도록 중점적으로 피드백받고 싶은 내용을 공유해주세요.  
예를 들어, 가장 고민했던 점이나 여전히 어려운 부분, 그리고 이에 대한 생각을 적을 수 있습니다. -->

안녕하세요 제이!
이번 출석 미션 제이의 리뷰이, 시소입니다!

아직 많이 부족하여 제가 작성한 코드도, 질문도 이해가 잘 안 가실 수 있을 것 같아요...
따끔하게 느낀 부분 있는 그대로 말씀해주시면 많이 배우겠습니다! ㅜ ㅜ 😿 

1. TDD를 처음 진행하여 과정이 손에 익지 않아 많은 어려움이 있었어요...
정의한 기능 요구사항에 대해 실패하는 테스트 코드를 작성하고, 성공할 수 있도록 로직을 작성, 성공한다면 리팩터링 하는 방식으로 배웠고, 이를 초반에는 잘 적용하며 진행했다고 생각했는데, 어느 시점에서 '여기서는 어떻게 테스트 코드를 작성하고 계속 진행하지..?' 생각이 드는 지점이 있었어요. 무언가 입력에 관한 부분이 연관된 지점에서 이런 기분을 느꼈는데요.

여기서 입출력에 관련된 부분을 TDD에서 테스트 할 필요가 있나요?
만약 필요하다면 어떤 방식으로 테스트를 해야 할까요?

2. 요구사항에서 제시한 2024년 12월이라는 기간이 현재보다 과거이기에 프로그램을 실행할 때에도, 테스트를 할 때에도 실행 기준 시점을 커스텀하여 필요하다고 판단하였어요. 이에 임시로 DateUtil이라는 날짜 관련 유틸 클래스를 만들어 그 내부에 TODAY라는 정적 변수로 임의로 실행 기준일을 정의해두고 전체적으로 해당 값을 사용하였는데요, 이 방식이 느낌상은 좋지 않은 방법이야...! 하고 경고하고 있는데 실제로도 좋지 않은 건지, 아니면 잘못된 방식이라고 볼 필요는 없는지 잘 판단이 가지 않아요. 만약에 잘못된 방식이라면 제가 어떤 방식으로 사용하는 게 좋을까요?

3. 처음 TDD를 진행하는 제게 당시 제이가 TDD를 처음 진행하며 느꼈던 고려할만한 팁이라던지, 아니면 이 부분은 꼭 지켰으면 좋겠다 하는 부분이 있을까요?

마감 기한을 지키느라 리팩터링을 충분히 챙기지 못하고, 객체 지향적인 부분을 충분히 고려하지 못한 상태로 제출을 하게 되어 너무 부끄럽지만,,, ㅜ ㅜ 이후에 다시 확인하면서도 제가 궁금한 것 추가적으로 여쭤볼게요!

이번 미션 잘 부탁드려요 감사합니다 ~~ 😸 

프롤로그 https://prolog.techcourse.co.kr/studylogs/3963

jjunh0 committed:
docs: 기능 목록 작성

lifeishiphop committed:
feat: 출석 기능 구현

- 등교 시간에 따라 출석, 지각, 결석을 판단하는 메서드를 구현하였습니다.

jjunh0 committed:
feat: 출석 기능 추가 구현

- 해당 날짜에 출석한 시간을 기록하는 로직을 추가하였습니다.

lifeishiphop committed:
feat: 출석 수정 기능 구현

- 해당 날짜에 대한 출석 시간을 수정하고, 해당 날짜에 대한 출석이 존재하지 않을 경우 예외가 발생하도록 구현하였습니다.

jjunh0 committed:
refactor: 책임에 따라 클래스 분리

- Crew, Crews 객체 추가
- attendanceBook 분리

lifeishiphop committed:
refactor: 크루, 출석 상태 클래스 및 테스트 리팩터링

- 리팩터링을 진행하였으며, 리팩터링 진행에 따라 테스트 코드도 수정하였습니다.

lifeishiphop committed:
docs: 예외 상황 추가

- 미래 날짜를 출석, 수정할 경우 예외 발생 기능을 추가하여 문서를 수정하였습니다.

jjunh0 committed:
feat: 결석 횟수 계산 로직 추가

- 공휴일 확인, LocalDate와 LocalTime을 합치기 위한 DateUtil 추가
-  주말을 확인하는 값을 enum으로 변경
- 결석 횟수 확인 테스트 코드 추가

lifeishiphop committed:
Merge branch 'step1' of https://github.com/jjunh0/java-attendance into step1

jjunh0 committed:
refactor: 조건문 메서드로 분리

lifeishiphop committed:
Merge branch 'step1' of https://github.com/jjunh0/java-attendance into step1

lifeishiphop committed:
feat: 실행 날짜 custom 지정 기능 추가

- 요구사항에서 필요한 날짜가 실제 기준으로 과거이기 때문에 날짜 유틸 클래스를 통해 실행 기준일을 별도로 지정하였습니다.

lifeishiphop committed:
feat: 지각, 결석 횟수 판별 기능 추가

- 각 크루별 지각, 결석 횟수 판별 메서드를 분리하여 계산하였습니다.

lifeishiphop committed:
feat: 제적 위험자 확인 기능 추가

- 제적 위험 상태 클래스를 ENUM으로 설계하고, 각 상태를 구분하는 계산 메서드를 작성하였습니다.

jjunh0 committed:
feat: 입력, 출력에 대한 부분 추가

- 사용자에게 메시지 출력과 입력 받는 부분 추가
- 요일 출력을 위한 ViewUtil 추가

lifeishiphop committed:
feat: 파일 입출력 기능 구현

- csv 파일을 통해 최초 입력을 수행하는 기능 작성 중에 있습니다.

lifeishiphop committed:
feat: 프로그램 실행 위한 Application 작성

jjunh0 committed:
feat: 출석 수정 입력시 출석 수정 로직 실행 기능 추가

- "2", "3" 입력시 정해진 로직 실행
- 입력 파일에 typo 수정

lifeishiphop committed:
feat: 크루별 출석 기록 확인 기능

- 크루별 출석 기록 확인 기능 작성 중에 있습니다.

lifeishiphop committed:
refactor: DateUtil 클래스 패키지 변경

- Domain 이외에도 공통적으로 사용하게 되어 global 패키지로 이동하였습니다.

lifeishiphop committed:
feat: 에러 메시지 추가

- 각 기능별 에러 발생 시 메시지 출력하는 기능을 추가하였습니다.

lifeishiphop committed:
feat: 출석일 판별 및 출석 기록 출력 기능 추가

- 공휴일, 주말을 제외한 출석일을 판별하는 메서드를 추가하였고, 출석 기록을 출력하는 기능을 구현 중에 있습니다.

jjunh0 committed:
feat: 출석일 판별 및 기록 출력 기능 완성

- 주말, 평일, 공휴일 판별에 대한 테스트 코드 작성
- printCrewAttendanceRecord 로직 분리
- Crew에대한 Dto 반환 로직 추가

lifeishiphop committed:
feat: 크루별 출석 횟수 및 제적 위험자 대상 여부 출력 기능 추가

jjunh0 committed:
refactor: 메서드 분리

- domain에 있던 view로직을 분리
- 메서드 명을 직관적으로 수정

jjunh0 committed:
feat: 제적 위험자 출력을 위한 로직 추가

jjunh0 committed:
test: 테스트 코드 작성

lifeishiphop committed:
feat: 예외 발생 기능 추가

- 예외 발생 및 메시지 기능을 추가하였습니다.

jjunh0 committed:
feat: 수정할때 수정 기록 메시지 출력 로직 추가

jjunh0 committed:
style: 코드 컨밴션

jjunh0 committed:
docs: 완료된 요구사항 체크

jjunh0 committed:
test: private에 대한 테스트 코드 삭제

lifeishiphop committed:
test: 확인하지 않은 테스트 추가

- 예외 발생 검증 기능에 대한 테스트를 수행하지 않아 추가하였습니다.

jjunh0 committed:
test: Crew, AttendanceStatus에 대한 테스트 코드 작성

JunHoPark93 commented on src/main/java/view/InputView.java:
```diff
@@ -0,0 +1,80 @@
+        this.scanner = scanner;
+    }
+
+    public String inputMenu() {
```
> 입출력에 관련된 부분을 TDD에서 테스트 할 필요가 있나요?
> 만약 필요하다면 어떤 방식으로 테스트를 해야 할까요?

작성주신 도메인들에 대한 테스트는 꼭 해야하지만 개인적인 생각으로 view같은경우는 굳이 tdd로 해야되나 라는 생각이 있네요..

lifeishiphop replied:
그렇군요 ㅎ 이전 프리코스 진행할 때에는 입출력에 관한 부분도 예제 테스트로 주어졌던 것 같아서 궁금했었어요!

JunHoPark93 commented on src/main/java/global/util/DateUtil.java:
```diff
@@ -0,0 +1,41 @@
+import java.time.LocalTime;
+
+public class DateUtil {
+    public static LocalDateTime TODAY = LocalDateTime.of(LocalDate.of(2024, 12, 17),
```
> 요구사항에서 제시한 2024년 12월이라는 기간이 현재보다 과거이기에 프로그램을 실행할 때에도, 테스트를 할 때에도 실행 기준 시점을 커스텀하여 필요하다고 판단하였어요. 이에 임시로 DateUtil이라는 날짜 관련 유틸 클래스를 만들어 그 내부에 TODAY라는 정적 변수로 임의로 실행 기준일을 정의해두고 전체적으로 해당 값을 사용하였는데요, 이 방식이 느낌상은 좋지 않은 방법이야...! 하고 경고하고 있는데 실제로도 좋지 않은 건지, 아니면 잘못된 방식이라고 볼 필요는 없는지 잘 판단이 가지 않아요. 만약에 잘못된 방식이라면 제가 어떤 방식으로 사용하는 게 좋을까요?

~FIXED등의 네이밍으로 의도를 더 나타내면 어떨까요?
요구사항에 `시범적으로 최소한의 기능을 갖춘 출석 시스템을 개발..`로 나와있기떄문에 시범기능에 대한 요구사항은 만족시키기떄문에 괜찮을거같아요.

lifeishiphop replied:
방법 자체는 요구사항을 만족시키기 때문에 문제가 없군요!
조금 더 의도를 드러낼 수 있도록 네이밍을 수정해볼게요~

JunHoPark93 commented on src/main/java/controller/AttendanceController.java:
```diff
@@ -0,0 +1,102 @@
+        outputView.printAttendDateAttendanceMessage(TODAY.toLocalDate(), crews.createCrewResponse(crew).attendanceBook());
+    }
+
+    private void editAttend() {
```
<img width="414" alt="스크린샷 2025-02-21 오후 11 32 54" src="https://github.com/user-attachments/assets/1850c07c-6325-4d3b-aa9f-8b176a21b7f9" />

출석수정이 의도대로 동작하지 않는것같은데 확인 가능하실까요?

lifeishiphop replied:
페어 프로그래밍 진행할 때 분명 해당 부분을 고려했다고 생각했었는데 어느 순간 놓쳤나봐요...
이런 부분은 조금 더 신경 써서 피드백 요청드릴게요 😿 

현재는 수정된 상태입니다!

JunHoPark93 commented on src/main/java/domain/AttendanceStatus.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    private static AttendanceStatus getAttendanceStatusByTime(LocalTime attendanceTime, LocalTime targetTime) {
+        if (targetTime.isAfter(attendanceTime)) {
+            if (attendanceTime.plusMinutes(30).isBefore(targetTime)) {
+                return ABSENCE;
```
`indent(인덴트, 들여쓰기) depth를 2를 넘지 않도록 구현한다. 1까지만 허용한다`
를 만족시키려면 어떻게 수정하면 좋을까요?

lifeishiphop replied:
내부 조건을 한 메서드로 빼어 분기를 나누는 방법이 있을 것 같아요.

당시에는 급하게 마감 기한을 맞추며 기능 구현을 우선시하여 지키지 못하였는데,
지금은 기존 코드에서 생각한 방식대로 한 번 수정해보았습니다!

JunHoPark93 commented on src/main/java/domain/RiskStatus.java:
```diff
@@ -0,0 +1,30 @@
+    public static RiskStatus getRiskStatus(int absenceCount, int tardyCount) {
+        int totalAbsenceCount = absenceCount + tardyCount / 3;
+
+        if (totalAbsenceCount > 5) {
+            return RiskStatus.EXPULSION;
+        }
+
+        if (totalAbsenceCount >= 3) {
+            return RiskStatus.COUNSELING;
+        }
+
+        if (totalAbsenceCount == 2) {
+            return RiskStatus.WARNING;
+        }
```
enum을 사용해주셨는데요, 메서드내에서 숫자를 써서 비교하지말고 각 상태별로 숫자값을 같이 가지고있도록 수정해보는건 어떨까요?

lifeishiphop replied:
지금보다 좋은 방법인 것 같아요! 반영해보겠습니다 😸

JunHoPark93 commented on src/main/java/domain/Crew.java:
```diff
@@ -0,0 +1,131 @@
+        attendanceBook.put(date, time);
+    }
+
+    public boolean isNameMatch(final String name) {
+        return this.name.equals(name);
+    }
```
작성주신 public메서드들에 대한 테스트는 전부 작성부탁드려요!

lifeishiphop replied:
헉 빠진 부분이 있었네요... 이 부분도 반영하겠습니다!

한 가지 궁금한 점이 있다면, 이렇게 테스트를 놓친 메서드가 있는지 확인하실 때 테스트 커버리지를 비교하며 확인하시는 건가요?

저는 커버리지에서 라인 번호에 표시된 색을 보고 확인하곤 하는데,
지난 미션 때도 다 시행했다고 생각했는데 하나씩 빼먹은 부분이 있더라구요... 😿

JunHoPark93 replied:
커버리지 수행을 해도되고 메서드 사용처 검색으로 보기도합니다! 혹은 주요메서드니 사용을 하면서 테스트가 있는지 확인하기도하고요

JunHoPark93 commented on src/main/java/domain/Crew.java:
```diff
@@ -0,0 +1,131 @@
+    private int calculateAbsenceCount() {
+        LocalDate localDate = DateUtil.getFirstDateOfMonth();
+        int absenceCount = 0;
+        while (!localDate.isAfter(TODAY.toLocalDate())) {
+            if (isNowAbsence(localDate)) {
+                absenceCount++;
+            }
+            localDate = localDate.plusDays(1);
```
이런 부분들도 마찬가지로 depth를 줄여볼까요?

lifeishiphop replied:
이 부분도 마찬가지로 depth 줄여두었습니다!

Stream을 사용하는 것 이외에 다른 방안은 떠오르지 않더라구요...
혹시 다른 방안이 떠오르신다면 추가로 알려주실 수 있을까요?!

JunHoPark93 replied:
스트림을 사용하거나 메서드로 추출해도 되지 않을까요? 
(저도 가끔은  2depth가 나아보일때도 있는데 지금은 요구사항에 맞는 극단적인 연습을 해보시는걸 추천드려요. 한번 해보고나면 추후에 기준 등이 생기실수도있어요)

JunHoPark93 commented on src/main/java/controller/AttendanceController.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    private void editAttend() {
+        String name = inputView.inputEditCrewName();
+        Crew crew = crews.findCrewByName(name);
+        LocalDate date = DateUtil.getDateByInputDay(Integer.parseInt(inputView.inputEditDay()));
+        validateIsFutureDate(date);
+        LocalTime beforeTime = crew.getAttendanceTime(date);
+        AttendanceStatus beforAttendanceStatus = crew.getAttendanceStatusByDate(DateUtil.assembleDateAndTime(date, beforeTime));
+        LocalTime time = LocalTime.parse(inputView.inputEditTime());
+        AttendanceStatus afterAttendanceStatus = crew.editAttendStatus(assembleDateAndTime(date, time));
+        LocalTime afterTime = crew.getAttendanceTime(date);
+        outputView.printAttendEditMessage(date, beforAttendanceStatus, beforeTime, afterAttendanceStatus, afterTime);
```
정답은없지만, 컨트롤러 내부에 로직들이 많이 존재하는데요
적절한 역할을 가진 도메인으로 옮겨볼 수 있을까요?

lifeishiphop replied:
이 부분은 사실 작성하면서도 어떻게 리팩터링을 해야 할지 고민했던 부분인데요,

로직을 분리하게 된다면 어떻게 하면 좋을지 피드백 받고 내내 구상을 해보았지만, 무언가 자연스럽다 싶은 구조가 잘 떠오르더라구요 ㅜ

여러 가지 방법을 고려해서 일단 지금까지 떠오른 방법 중에 그나마 낫다고 생각한 방법을 적용해봤는데요.

수정 전 시간과 상태를 가져오고, 수정 후 시간과 상태를 가져오는 로직을 한 가지로 합치게 되었고,
날짜에 대한 검증을 Crew 클래스 로직 내부에서 수정이 가능한 날짜인지 체크할 때 확인하도록 수정해보았어요.
이런 부분에서 수정 전후의 시간과 상태를 함께 가져오도록 DTO를 사용을 고려하게 되었는데요.

여전히 많은 로직이 존재하는 것 같지만... 나은 방법을 더 고려해보겠습니다 😿 

추가로 DTO 사용과 관련하여 여쭤보고 싶은 부분이 있어 comment 따로 남겨두겠습니다!

JunHoPark93 reviewed (changes requested):
안녕하세요 시소~
1단계 구현 하느라 고생많으셨습니다.
몇가지 코멘트 남겨두었는데 확인 후 의견 및 반영부탁드립니다~

JunHoPark93 commented:
> 처음 TDD를 진행하는 제게 당시 제이가 TDD를 처음 진행하며 느꼈던 고려할만한 팁이라던지, 아니면 이 부분은 꼭 지켰으면 좋겠다 하는 부분이 있을까요?

작성주신 도메인들에 대한 모든 메서드들에 대해서 테스트가 되었으면 좋겠습니다. 팁은 글쎄요 발생할 수 있는 다양한 경우를 생각해보는것이 있을거같은데요, 오히려 view테스트 이외에 어려웠던 점을 말씀주시면 그 부분에 대해서 이야기를 해보면 좋을거같습니다.

lifeishiphop committed:
test: 이름 일치 확인 메서드에 대한 테스트 추가

lifeishiphop committed:
refactor: 기준일 상수 네이밍 변경

lifeishiphop committed:
refactor: import문 컨벤션 수정

lifeishiphop committed:
refactor: 매직 넘버 추출 및 출석 판별 메서드 리펙터링

- 가독성을 위해 상수를 추출하고, depth를 1로 줄이고자 메서드를 분리하였습니다.

lifeishiphop committed:
fix: 출석 수정 시 예외 발생 관련 버그 수정

- 출석 기록이 없는 날짜에 대한 출석 수정 시도에서 날짜와 시간 모두 입력되어야 예외가 발생하는 버그를, 날짜만 입력되어도 즉시 예외가 발생하도록 수정하였습니다.

lifeishiphop committed:
refactor: 제적 위험도 클래스의 각 상태가 판별값을 지니도록 수정

lifeishiphop committed:
fix: 출석 상태를 제대로 반환하지 않는 버그 수정

lifeishiphop committed:
refactor: Controller의 출석 수정 로직 해당 도메인으로 분리

- 출석 시간과 상태를 DTO에 매핑하도록 수정하였습니다.

lifeishiphop committed:
refactor: 불필요한 주석 및 미사용 메서드 제거

lifeishiphop committed:
refactor: Validator 로직을 각 책임에 따라 분리

egaeng09 commented:
추가적으로 여쭤보고 싶은 부분이 있어 질문 남겨요!

1. DTO 사용에 관련하여, 저는 사실 지난 미션까지도 DTO를 사용할 이유를 찾지 못했어요.     

하지만 **이번 미션에서는 도입해도 좋겠다**는 생각을 하였는데요, 그 **근거**로는 각 계산된 값들을 Controller에서 일일이 저장해두고 View로 넘겨주게 되면 불필요한 코드가 늘어난다고 생각했고, 추가적으로 필요한 값들이 대부분 제 프로그램 설계 상 객체의 상태로써 저장하기보다 일일이 계산하는 것이 대부분이었기에 이 부분 하나 때문에 객체를 추가적으로 만들어 매 로직이 실행될 때마다 갱신되도록 만들 필요가 있을까? 생각했기 때문이에요. **즉, 도메인이 가지고 있는 상태를 그대로 뷰에서 사용하는 것이 아닌 가공(?) 계산(?)된 값을 필요로 하기 때문이라고 말할 수 있을 것 같아요.**    

이전 미션의 제 리뷰어였던 베루스는 이전 미션에서 DTO를 사용하는 것에 대해 무조건적으로 부정적인 견해를 가지고 있지는 않았지만, 다른 크루들의 리뷰어들은 종종 DTO를 사용하지 마라는 피드백을 받곤 한다더라구요. 저는 반대로 DTO를 사용하지 않을 필요까지야 있을까? 하는 생각이 들어서요.

제이가 바라본 이번 미션에서의 DTO 사용에 대한 의견을 묻고 싶어요.

<br/>

2. 다른 도메인보다 Crew에 더 많은 로직들이 들어가있지 않은가? 하는 생각이 들어요.

사실 어떻게 보면 출석(날짜, 시간)에 관련된 상태를 지니고 있기 때문에 이정도면 적절하지 않을까? 생각이 들기도 하구요.
Crew에 조금 가중된 책임을 부여한 것 같나요?

<br/>

3. 만약 제가 메뉴 선택에 관한 ENUM 클래스를 만들어 각 1, 2, 3, 4, Q를 상태로써 가지고, 선택 메뉴를 판별하는 of 메서드를 내부에 만들도록 설계한다면 해당 ENUM 클래스는 view 패키지 내부에 존재해야 하나요?

저는 `선택한 메뉴를 통해 메뉴가 무엇인지 판단하는 로직`이 있다고 생각하여 단순히 domain에 있어야 하지 않을까 생각했는데, 어떻게 보면 1, 2, 3, 4, Q를 구분하여 메뉴를 나누는 것은 View에서 필요한 로직이기 때문에 view 패키지에 있어야 하나? 하는 생각이 들더라구요.

<br/>

의도한대로 작동하지도 않았고, 테스트도 빼먹는 엉터리 리뷰 요청이었지만 상세하게 잘 봐주셔서 감사해요! 😸 
어쩌다보니 벌써 월요일이 지났는데, 이번 한 주도 파이팅이에요 ~~ 👊 👊

JunHoPark93 commented on src/main/java/domain/Crew.java:
```diff
@@ -0,0 +1,138 @@
+        return RiskStatus.getRiskStatus(calculateAbsenceCount(), calculateTardyCount());
+    }
+
+    public CrewResponse createCrewResponse() {
```
> DTO 사용에 관련하여, 저는 사실 지난 미션까지도 DTO를 사용할 이유를 찾지 못했어요.

전달할 인자값들이 매우 많은경우는 클래스로 만드는게 더 나을수도있습니다. 하지만 dto사용이 무조건 나쁘다기보다 클라이언트 요구사항에 도메인 클래스가 영향을 받기 때문에 이러한 현재 구조는 지양하는걸 추천드려요

JunHoPark93 commented:
> 다른 도메인보다 Crew에 더 많은 로직들이 들어가있지 않은가? 하는 생각이 들어요.

정답은 없지만 (어렸을때 학원가면 출석담당하는 사람이 따로 있는경우도있는것처럼..) 출석자체에 관련된 내용은 다른 도메인에게 부여해볼 수도 있을거같네요.

JunHoPark93 commented:
> 만약 제가 메뉴 선택에 관한 ENUM 클래스를 만들어 각 1, 2, 3, 4, Q를 상태로써 가지고, 선택 메뉴를 판별하는 of 메서드를 내부에 만들도록 설계한다면 해당 ENUM 클래스는 view 패키지 내부에 존재해야 하나요?

넵 저도 말씀하신 내용에 동의합니다. view에서만 쓰이는거면 view패키지 내부에 존재하는것이 맞을거같아요

JunHoPark93 reviewed (approved):
안녕하세요 시소~
리뷰 반영하시느라 고생많으셨습니다.
혹시라도 해결되지 않은게 있다면 다시 말씀부탁드려요!
다음 단계에서 뵐게요
